### PR TITLE
15 - Add the CI setup with GitHub actions

### LINKED ISSUE #15: CI setup with Github Actions
## Context
Code changes should be validated running the test suite every time a pull request is opened 

## Objective
* To implement a CI server with Github actions

### PR DESCRIPTION
**Pull request by juliansg022**
Closes #15 
Hi @juanhurtado10. I added the setup file to run the tests on each PR.

### CODE REVIEW

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ main, v* ]
+  pull_request:
+    branches: [ '**' ]
```
[the remark this answers is not in the document]
Comment (juliansg022) [reply .github/workflows/node.js.yml:10]: @juanhurtado10 Could I then omit lines #9 and #10, or just #10?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ main, v* ]
```
[the remark this answers is not in the document]
Comment (juliansg022) [reply .github/workflows/node.js.yml:8]: @juanhurtado10 You're right, it's a better format

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run build --if-present
```
[the remark this answers is not in the document]
Comment (juanhurtado10) [reply .github/workflows/node.js.yml:30]: @juliansg022 let's keep it as simple as possible.

File: .github/workflows/node.js.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanhurtado10) [reply .github/workflows/node.js.yml:10]: @juliansg022 just the line#10

**Comment by juliansg022** [comment]
@juanhurtado10 The suggested corrections were included and the bug reported by GitHub was fixed.

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**